pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -40,7 +41,9 @@ type PodCache struct {
 	// pod cache if a pod changes IP.
 	IPByPods map[string]string
 
-	// needResync is map of IP to endpoint namespace/name. This is used to requeue endpoint
+	podTransitionTime map[string]kubemetav1.Time
```

Comment:
This is not obvious, could use a comment about what it is, why we need it, and what the key is (since its just "string")